### PR TITLE
IBX-6644: Gracefully handle broken custom URL aliases

### PR DESCRIPTION
**Pull request by barw4**
| Question      | Answer
| ------------- | ---
| Tickets       | [IBX-6644](https://issues.ibexa.co/browse/IBX-6644)
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

Try/catch was added with logging.

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

File: src/bundle/Resources/views/themes/admin/content/tab/url/custom_urls_table.html.twig [hunk 1]
```diff
@@ -45,7 +45,7 @@
                 <tr>
                     <td class="ez-table__cell ez-table__cell--no-content" colspan="4">
                         <p class="ez-table-no-content mb-0 py-0">
-                            {{ 'tab.urls.no_custom_urls'|trans|desc('This Content item has no custom URL aliases.') }}
+                            {{ 'tab.urls.no_custom_urls'|trans|desc('This Content item has no custom URL aliases or they are broken.') }}
```
Comment (Steveb-p) [inline src/bundle/Resources/views/themes/admin/content/tab/url/custom_urls_table.html.twig]: Pass this message to people handling microcopy - I'm expecting them to not be happy about always suggesting that some links are broken.

File: src/lib/UI/Dataset/CustomUrlsDataset.php [hunk 2]
```diff
@@ -43,7 +50,24 @@ public function __construct(
...
+                true
+            );
+        } catch (BadStateException $e) {
+            $this->logger->warning(
+                sprintf(
+                    'At least one custom alias belonging to location %d is broken. Fix it by using the ezplatform:urls:regenerate-aliases command.',
+                    $location->id
+                ),
+                [$e->getMessage()]
+            );
```
Comment (Steveb-p) [inline src/lib/UI/Dataset/CustomUrlsDataset.php]: Context is passed incorrectly.
```suggestion
            $this->logger->warning(
                sprintf(
                    'At least one custom alias belonging to location %d is broken. Fix it by using the ezplatform:urls:regenerate-aliases command.',
                    $location->id
                ),
                ['exception' => $e]
            );
```

File: src/bundle/Resources/config/services.yaml [hunk 3]
```diff
@@ -59,6 +59,8 @@ services:
         lazy: true
         arguments:
             $configResolver: '@ezpublish.config.resolver'
+        calls:
+            - [setLogger, ['@?logger']]
```
Comment (Steveb-p) [inline src/bundle/Resources/config/services.yaml]: With proper autowiring/autoconfiguration this method should be called by Symfony framework on it's own.

Consider adding a `monolog.logger` tag with `ibexa.admin` channel.

**Review by ViniTou — APPROVED** [review]

**Review by konradoboza — APPROVED** [review]

**Review by Steveb-p — APPROVED** [review]

**Review by tomaszszopinski — APPROVED** [review]
QA approved on IbexaDXP 4.6 commerce.

**Pull request merged**